### PR TITLE
adding mysqlreceiver

### PR DESCRIPTION
**Pull request by samiura**
**Description:** Adding native mysql receiver.

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: docs/components.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ The distribution offers support for the following components.
 | [lightprometheus](../internal/receiver/lightprometheusreceiver)                                                                         | [in development] |
 | [mongodbatlas](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/mongodbatlasreceiver)               | [beta]           |
 | [mongodb](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/mongodbreceiver)               | [beta]           |
+| [mysqlreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/mysqlreceiver)                     | [stable]         |
```
Comment (crobert-1) [inline docs/components.md:37]:
```suggestion
| [mysqlreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/mysqlreceiver)                     | [beta]         |
```
I believe it's [still beta for metrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/mysqlreceiver)

File: tests/tools/go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module github.com/signalfx/splunk-otel-collector/tests/tools
 
-go 1.20
+go 1.21
+
+toolchain go1.21.1
```
Comment (crobert-1) [inline tests/tools/go.mod:6]: This should probably be done in another PR (unless I'm missing something)

File: docs/components.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rmfitzpatrick) [reply docs/components.md:37]: also this is in beta

File: tests/tools/go.mod (hunk 2 above)
Comment (samiura) [reply tests/tools/go.mod:6]: It was auto generated for my machine when I did go tidy I believe.

File: go.mod [hunk 3]
```diff
@@ -175,7 +176,7 @@ require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/internal/kafka v0.89.0 // indirect
 	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/batchpersignal v0.89.0 // indirect
 	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl v0.89.0 // indirect
-	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatautil v0.89.0 // indirect
+	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatautil v0.90.0 // indirect
```
[the remark this answers is not in the document]
Comment (samiura) [reply go.mod:179]: okay.

File: tests/tools/go.mod (hunk 2 above)
Comment (rmfitzpatrick) [reply tests/tools/go.mod:6]: I think if you undo the mod changes, `export GOTOOLCHAIN=local`, and make tidy it won't restrict for others?

we probably need to add this as a default env var for make.

File: tests/tools/go.mod [hunk 4]
```diff
@@ -2,6 +2,7 @@ module github.com/signalfx/splunk-otel-collector/tests/tools
 
 go 1.20
 
+
```
Comment (crobert-1) [inline tests/tools/go.mod:5]:
```suggestion
```

File: go.mod (hunk 3 above)
[the remark this answers is not in the document]
Comment (crobert-1) [reply go.mod:179]: @rmfitzpatrick: Out of ignorance, should other dependencies remain at `v0.89.0` for now as well? They'll be updated soon automatically, but not sure what the best practice is here.

**Comment by atoulme** [comment]
Please rebase, and update to 0.90.0.
